### ISSUE TITLE
[Bug] Long text via cat makes the tts engine mumble after a while

### ISSUE THREAD

**Issue by domasofan** [rhasspy/piper#211]
Tested on Windows 10 64 bit and Piper 2023.9.9-1 prerelease.
A ported cat was used.

When a text has around 6000 characters piper starts mumbling after reading a while.
Command used was:
$ cat demo.txt | piper -m model.onnx -f demo.wav

Model used was de_DE_thorsten_high.onnx
Though i guess it can be tested with other models as well.

**Comment by domasofan** [comment]
Just tested with latest release 2023.11.14-2.
Tested with a long 23:29 minute file.
The problem has been fixed or fixed itself in parts.
Mumbling/slurring happens randomly from time to time.
First time after speaking more than 2 minutes.
Happened here with the test file at the following times:
2:10-2:40, 11:59-12:14, 14:33-15:12.

The demo MP3 can be found here:
https://cloud.hohenems.at/index.php/s/867FWLXRmwwQzCA

The demo text used is here:
https://cloud.hohenems.at/index.php/s/ePpj7bDnzT7J58a

It seems to recover after 20-30 seconds.

Tested with the de_DE-thorsten-high voice.

**Comment by clort81** [comment]
I am running repeated invocations with de_DE-thorsten-high and about one in 5 times it speaks at double speed - not exactly mumbling, but everything is choppy and average word per minute doubles.

I have not found a problem with any other model, yet.  This may be a Thorsten-only issue.

Please inform if you have found the described issue with other voices.

**Comment by domasofan** [comment]
I guess this is not affected by cat input
Seems this is resulting from piping raw output into ffmpeg or lame.
If i just let it write into wav files directly i don't get any errors.
de_DE-mls-medium seems to be affected most.
this is a multi speaker model. i used speaker 1.
Tested under Windows.
Can't currently test on Linux.

**Comment by Moonbase59** [comment]
Using piper-tts 1.2.0 (installed using pip3), the `de_DE-mls-medium` model produces unusable, outlandish sounding output in 90% of cases for me. Interestingly enough, the "Rainbow" test sentence from the samples web page always works.

Here is an example output of `de_DE-mls-medium`, speaker 3, speaking the sentences:

- _Dies ist ein Test._
- _Der Regenbogen ist ein atmosphärisch-optisches Phänomen, das als kreisbogenförmiges farbiges Lichtband in einer von der Sonne beschienenen Regenwand oder -wolke wahrgenommen wird._

[de_DE-mls-medium.Testfiles.zip](https://github.com/user-attachments/files/15520527/de_DE-mls-medium.Testfiles.zip)

Used
```
echo "Dies ist ein Test." | piper -m /home/matthias/.local/share/piper/voices/de_DE-mls-medium.onnx -s 3 --length-scale 1 -f /tmp/jingle.wav
```
to produce these (I have my voices in `~/.local/share/piper/voices/`). Same result when not using `--length-scale`.

**Comment by TikoTako** [comment]
I don't speak german but can understand some words so i did few tests:

Test 1: 
`echo "oktoberfest brezel eins zwei polizei guten tag" | piper -m de_DE-mls-medium.onnx -s 3 --length-scale 1 --output-raw | ffplay -f s16le -ar 22050 -i pipe: -autoexit`

Test 2: 
`echo "oktoberfest brezel eins zwei polizei" | piper -m de_DE-mls-medium.onnx -s 3 --length-scale 1 --output-raw | ffplay -f s16le -ar 22050 -i pipe: -autoexit`

I ran the test picking random speakers, _de_DE-mls-medium_ need a minimum of few words to work correctly, Test 1 sounds fine while Test 2 sound weird, going down is even worse.

I tested another model, _de_DE-thorsten-medium_, it work fine even with only one word.

**Comment by Moonbase59** [comment]
Yeah, it seems the multi-speaker (most versatile) voices don’t work correctly. _de_DE-thorsten_ is always good.
